Sync Modal: improve layout and spacing for better fit on smaller screens

katinthehatsite opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Closes DOTCOM-14140

## Proposed Changes

This PR reduces spacing around the Sync modal to make the items a bit more compact and smaller screen friendly. 

**Before**

<img width="1462" height="861" alt="Screenshot 2025-08-07 at 3 15 49 PM" src="https://github.com/user-attachments/assets/6cc3d951-ca07-46bb-addd-6928fb02ddb5" />

**After**

<img width="1462" height="904" alt="Screenshot 2025-08-07 at 3 15 26 PM" src="https://github.com/user-attachments/assets/7353c622-4be4-4642-a0f4-8c6cc7f20b8e" />

Observe that we have no scrollbar here and all the content fits when you open at least the first set of options. 

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Pull the changes from this branch or use the Calypso Live link below
* Ensure that you have an Atomi site and a staging site 
* Navigate to `/staging-site/{yourAtomicSite}`
* Click on the `Sync` button to open the sync modal
* Observe the spacing in the modal

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
Remove spacing

Commit:
Reduce bottom spacing

Commit:
Drop bottom border

katinthehatsite commented:
@epeicher @ndiego Please give it a test and let me know what you think, we can make further modifications. I see that we do not have any design input on this but I think we can move with small tweaks forward on our end.

epeicher commented:
I think this needs designer eyes @Marinatsu 👀 . I have tested it and the scroll starts at around 750px height as 

||Before | After|
|-------|-------|------|
|~ 750px height|<img width="3194" height="1500" alt="CleanShot 2025-08-07 at 18 09 16@2x" src="https://github.com/user-attachments/assets/2d7d73fa-68bc-4384-bbc0-5ce0f690529f" />| <img width="3194" height="1500" alt="CleanShot 2025-08-07 at 18 08 59@2x" src="https://github.com/user-attachments/assets/5446c101-9450-49a4-a2c6-4c324696c792" />|
|~1220px height  | <img width="3194" height="2442" alt="CleanShot 2025-08-07 at 18 09 41@2x" src="https://github.com/user-attachments/assets/fabc3bb8-49f4-48e0-bf9b-827654a2aa16" />| <img width="3194" height="2442" alt="CleanShot 2025-08-07 at 18 10 00@2x" src="https://github.com/user-attachments/assets/3736d88d-fbc4-421d-806b-7d3b9fe11765" />|

katinthehatsite commented:
Thanks for testing @epeicher ! I asked for the design feedback in p1754640213303199-slack-C04GESRBWKW .
Let's hope we can get further guidance and proceed from there.

Commit:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into fix/update-styles-on-modal

Commit:
Remove heading and adjust spacing

Commit:
Adjust spacing on the top

Commit:
Drop padding under warning

Commit:
Adjust spacing below the component

katinthehatsite commented:
Slack discussion can be found here: p1754640213303199-slack-C04GESRBWKW

Currently, I adjusted it to what is indicated on the screenshot and removed "What would you like to pull/ push?"

<img width="869" height="659" alt="Screenshot 2025-08-11 at 11 01 57 AM" src="https://github.com/user-attachments/assets/9548030b-0360-48d3-920b-28b3fcf2af99" />

ivan-ottinger reviewed (approved):
Nice! Thanks for experimenting with those design updates, Kat.

The changes look good to me: the code and also in how they are adjusting the modal display:

<img width="3224" height="2246" alt="Markup on 2025-08-11 at 12:35:57" src="https://github.com/user-attachments/assets/780a6b2f-c95d-4090-b192-2d3774ee6685" />

<img width="3244" height="2232" alt="Markup on 2025-08-11 at 12:34:28" src="https://github.com/user-attachments/assets/d2fd6032-2198-4fd7-8b13-aa3a19722d25" />

I don't miss the "What would you like to pull / push?" heading. The modal seem clear to me without it as well.

There's one thing that we may want to look into (maybe in a different PR). If the file tree is extended and we click the "Database Tables" checkbox, the warning is not visible:

https://github.com/user-attachments/assets/a19b97f3-cb44-4fe3-aeee-be277300234e

Maybe a possible solution could be to scroll to the bottom for the user automatically - or - place the warning together with the confirmation field below (sticky at the bottom).

Commit:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into fix/update-styles-on-modal

katinthehatsite commented:
> Maybe a possible solution could be to scroll to the bottom for the user automatically - or - place the warning together with the confirmation field below (sticky at the bottom).

Let's add an issue to check the best approach for this :) I will create it and follow up here.
